script.js

```diff
@@ -0,0 +1,365 @@
+  const filteredRecipes = recipes.filter((recipe) =>
+    recipe.title.toLowerCase().includes(wordForSearchInput)
+  );
+  
+  ingredientSearchInput.value = '';
+  renderRecipes(filteredRecipes);
+}
+
+// Find Ingredient By Word
+ingredientSearchInput.addEventListener('keydown', (event) => {
+  if (event.key === 'Enter') {
+    findRecipeByIngredient();
+  }
+});
+
+findRecipeByIngredient();
+function findRecipeByIngredient() {
+  searchingredientButton.addEventListener(`click`, findRecipeByIngredient);
+
+  const ingredientForSearchInput = ingredientSearchInput.value.trim().toLowerCase();
+  const filteredRecipes = recipes.filter((recipe) =>
+    recipe.ingredients.some((ingredient) =>
+      ingredient.name.toLowerCase().includes(ingredientForSearchInput)
+    )
+  );
+  searchInput.value = '';
+  renderRecipes(filteredRecipes);
+}
+
+function clearInputFields() {
+  searchInput.value = '';
+  ingredientSearchInput.value = '';
+}
+
+// sort By Amount of Ingredients
+let isSortedDescending = true;
+const sortButton = document.querySelector('#sort-button');
+if (sortButton) {
+  sortButton.addEventListener('click', sortByAmountIngredients);
+}
+
+function sortByAmountIngredients() {
+  recipes.sort((a, b) =>
+    isSortedDescending
+      ? b.ingredients.length - a.ingredients.length
+      : a.ingredients.length - b.ingredients.length
+  );
+  isSortedDescending = !isSortedDescending;
+  clearInputFields();
+  renderRecipes(recipes);
+}
+
+// Page-Timer
+function startPageTimer() {
+  const timerElement = document.querySelector('#page-timer');
+  if (!timerElement) {
+    console.error("Element with id '#page-timer' not found.");
+    return;
+  }
+
+  let seconds = 0;
+
+  function updateDisplay() {
+    const hours = Math.floor(seconds / 3600).toString().padStart(2, '0');
+    const minutes = Math.floor((seconds % 3600) / 60).toString().padStart(2, '0');
+    const secs = (seconds % 60).toString().padStart(2, '0');
+    timerElement.textContent = `${hours}:${minutes}:${secs}`;
+  }
+
+  setInterval(() => {
+    seconds++;
+    updateDisplay();
+  }, 1000);
+
+  updateDisplay();
+}
+
+startPageTimer();
+
+// Custom-Timer
+let isTimerRunning = false;
+let timerInterval = null;
+
+function startCustomTimer() {
+  const timerInput = document.querySelector('#timer-input').value.trim();
+  const timerDisplay = document.querySelector('#timer-display');
+  const timeSound = document.querySelector('#alarm-sound');
+  const finishSound = document.querySelector('#finish-sound');
+  const startButton = document.querySelector('#start-timer');
+
+  const minutes = parseInt(timerInput, 10);
+
+  if (isTimerRunning) {
+    stopTimer();
+    return;
+  }
+
+  if (isNaN(minutes) || minutes <= 0) {
+    alert('Please enter a valid time in minutes (greater than 0)');
+    return;
+  }
+
+  if(minutes > 600){
+    alert('Please enter a valid time in minutes (less than 10 hours)');
+    return;
+  }
+
+  let totalSeconds = minutes * 60;
```

Comment:
Another approach you can follow is to have new Date() object to track the date when you started visiting the website and then you compare it to the current date to get the duration (the time difference)